schemas/vulnerability.yml

```diff
@@ -0,0 +1,176 @@
+      level: extended
+      type: float
+      short: Vulnerability Environmental score.
+      description: >
+        Scores can range from 0.0 to 10.0, with 10.0 being the most severe.
+
+        Environmental scores cover an assessment for any modified Base metrics,
+        confidentiality, integrity, and availability requirements.
+        For example (https://www.first.org/cvss/specification-document)
+
+      example: 5.5
+
+    - name: score.version
+      level: extended
+      type: keyword
+      short: CVSS version.
+      description: >
+        The National Vulnerability Database (NVD) provides qualitative severity rankings
+        of "Low", "Medium", and "High" for CVSS v2.0 base score ranges in addition to the
+        severity ratings for CVSS v3.0 as they are defined in the CVSS v3.0 specification.
+
+        CVSS is owned and managed by FIRST.Org, Inc. (FIRST), a US-based non-profit
+        organization, whose mission is to help computer security incident response teams
+        across the world.
+        For example (https://nvd.nist.gov/vuln-metrics/cvss)
+
+      example: 2.0
+
+    - name: category
+      level: extended
+      type: keyword
+      short: Category of a vulnerabilty.
+      description: >
+        The type of system or architecture that the vulnerability affects. These may be
+        platform-specific (for example, Debian or SUSE) or general (for example, Database
+        or Firewall).
+        For example (https://qualysguard.qualys.com/qwebhelp/fo_portal/knowledgebase/vulnerability_categories.htm)
+
+      example: Firewall
+
+    - name: description
+      level: extended
+      type: keyword
+      short: Description of the vulnerability.
+      description: >
+        The description of the vulnerability that provides additional context of the
+        vulnerability.
+        For example (https://cve.mitre.org/about/faqs.html#cve_entry_descriptions_created)
+
+      example: In macOS before 2.12.6, there is a vulnerability in the RPC...
+
+    - name: scanner.id
+      level: extended
+      type: keyword
+      short: ID of the vulnerability.
+      description: >
+        The identification (ID) is the number portion of a vulnerability entry. It
+        includes a unique identification number for the vulnerability.
+        For example (https://cve.mitre.org/about/faqs.html#what_is_cve_id)
+
+      example: CVE-2019-00001
+
+    - name: scanner.vendor
+      level: extended
+      type: keyword
+      short: Name of the scanner vendor.
+      description: >
+        The name of the vulnerability scanner vendor.
+
+      example: Tenable
+
+    - name: severity
+      level: extended
+      type: keyword
+      short: Severity of the vulnerability.
+      description: >
+        The severity of the vulnerability can help with metrics and internal
+        prioritization regarding remediation.
+        For example (https://nvd.nist.gov/vuln-metrics/cvss)
+
+      example: Critical
```

Comment:
@randomuserid @dainperkins @MikePaquette @FrankHassanabad 

I'd like us to discuss `vulnerability.severity` a bit.

ECS already has a few fields close to this. I don't think we should use them, but perhaps we need to call out to them, one way or another.

- `event.severity` is numeric, so not a good fit for this.
- `log.level` and `event.severity` are meant to be equivalent, where `log.level` is the textual representation of `event.severity`. But I also wouldn't expect importing vulnerabilities and having their severity in `log.level`. So also a no-go.

But should we give instructions to e.g. copy `vulnerability.score.base` to `event.severity`? Any other opinions on this?

Reply:
Severities are generally calculated by the scanner through an internal algorithm. I can't say definitively if the "high" severity by Qualys is the same as the "high" from Tenable, but like the `vulnerability.category`, I think we should let the scanner decide on this vs. us maintaining some matrix.

Reply:
Totally agree on that point, I don't want to introduce a mapping of these values.

And likewise, if we were to instruct people to copy e.g. the base score out to a more central field like `event.severity`, the idea is still that the value is source-specific, not mapped.

But perhaps we shouldn't worry about this for now, and only address this if/when we have a concrete need for that. I looked around SIEM, and `event.severity` is not a default column anywhere that I could see.